### DISCUSSION TITLE
Question about the calculation of speed loss

### DISCUSSION THREAD

**Discussion by jiangshengkai1999** [NatLabRockies/floris#948] [Q&A] [answered]
I saw in the solver.py file that velocity_deficit is first calculated when calculating the wake loss for each wind turbine. For the wake model, velocity_deficit should be the velocity of the downstream affected by this wind turbine divided by the inflow velocity of this wind turbine. In floris, it is directly multiplied by flow_field.u_initial_sorted, which is the inflow velocity not affected by the wake. Is this reasonable?

**Comment by misi9170** [comment, marked as answer]
Hi @jiangshengkai1999 ,

Taking the [`sequential_solver()`](https://github.com/NREL/floris/blob/main/floris/core/solver.py#L48) as an example, the [`velocity_deficit`](https://github.com/NREL/floris/blob/main/floris/core/solver.py#L203) is, as you say, the speeds in the wake of turbine `i` normalized by the free stream velocity. These are then combined with other wakes in the [`wake_field`](https://github.com/NREL/floris/blob/main/floris/core/solver.py#L217). That is, the wake velocities are recovered using the free stream velocity [`flow_field.u_initial_sorted`], and _then_ combined with the other wakes to compute the combined effects of multiple wakes. The resulting `wake_field` (which is a velocity deficit in absolute values, i.e. in m/s) is then subtracted from the freestream wind field to compute [`flow_field.u_sorted`](https://github.com/NREL/floris/blob/main/floris/core/solver.py#L252), which is used as the inflow to downstream turbines for computing their [thrust coefficient](https://github.com/NREL/floris/blob/main/floris/core/solver.py#L92) and [axial induction factor](https://github.com/NREL/floris/blob/main/floris/core/solver.py#L114).

This is the way that most papers presenting analytical wake models compute the combined effects of multiple wakes.

**Reply by jiangshengkai1999** [reply]
@misi9170 Thank you for your answer, I completely understand what you mean and your answer is correct. The source of my question is that some papers give a square sum superposition model as shown in the picture, where the velocity loss uij is divided by the incoming flow velocity uj of the wind turbine. After discussing with others, we found that this is essentially consistent with the square sum model in Floris. I wonder if you have noticed this or if you have other ideas.
<img width="220" alt="屏幕截图 2024-07-24 135121" src="https://github.com/user-attachments/assets/2784c75d-8c37-4244-aae6-b9c4a722c025">

**Reply by misi9170** [reply]
Ah I see, interesting. Yeah, that form would seem to assume that the velocity incident to turbine $j$ ($u_j$) is a more appropriate representation of the "background" flow for the wake of turbine $j$ than the freestream velocity. I can see an argument for that, but that's not how things are implemented in FLORIS---generally, I think we could write the sum of squares method as

$$ \left(u_{\text{inflow}} - u_i\right)^2 = \sum_{j=1}^n \left(u_{\text{inflow}} - a u_{ij} \right)^2 $$

where, in the form of FLORIS and originally in [Katic et al (1987)](https://backend.orbit.dtu.dk/ws/portalfiles/portal/106427419/A_Simple_Model_for_Cluster_Efficiency_EWEC_86_.pdf), $a = 1$; whereas in the equation you've shown,
$a=\frac{u_{\text{inflow}}}{u_j}$ (Katic et al is not completely clear on this point, however, if turbine $j$ is not a freestream turbine).

Either way, the wake combination model should be treated as somewhat heuristic. [Gunn (2019)](https://iopscience.iop.org/article/10.1088/1742-6596/1222/1/012003) argues that the sum of squares model (using the freestream velocity, i.e. $a=1$) is a convenient heuristic that actually covers two effects: the expansion of the streamtube of the upstream wake due to the axial induction of the downstream turbine; and the "correct" momentum-conserving combination of the wakes (again, specified as deficits from the freestream velocity) downstream of the second turbine. I'm not sure exactly where the $a=\frac{u_{\text{inflow}}}{u_j}$ would fit in in that context.
